src/main/java/com/revature/controllers/QuestionController.java

```diff
@@ -40,4 +42,95 @@
+	public ResponseEntity<QuestionPool> updateQuestion(@RequestBody QuestionPool question, @PathVariable ("id") Integer id){
+		
+		QuestionPool currQuestion = questionLogic.getQuestionById(id);
+		
+		if(currQuestion == null){
+			System.out.println("Question with id " + id + " not found");
+	        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		}
+		
+		if(question.getQuestionText() != null){
+			currQuestion.setQuestionText(question.getQuestionText());
+		}
+		if(question.getMaxCommunicationScore() != null){
+			currQuestion.setMaxCommunicationScore(question.getMaxCommunicationScore());
+		}
+		if(question.getMaxKnowledgeScore() != null){
+			currQuestion.setMaxKnowledgeScore(question.getMaxKnowledgeScore());
+		}
+		if(question.getSubjectId() != null){
+			currQuestion.setSubjectId(question.getSubjectId());
+		}
+		if(question.getUseCount() != null){
+			currQuestion.setUseCount(question.getUseCount());
+		}
+		if(question.getDateLastUsed() != null){
+			currQuestion.setDateLastUsed(question.getDateLastUsed());
+		}
+		
+		return ResponseEntity.ok(questionLogic.updateQuestion(currQuestion));
+	}
+	
+	@RequestMapping(method = RequestMethod.DELETE, value = "/delete/{id}")
+	public ResponseEntity<QuestionPool> deleteEval(@PathVariable ("id") int id){
+		QuestionPool currQuestion = questionLogic.getQuestionById(id);
+		
+		if(currQuestion == null){
+			System.out.println("Question with id " + id + " not found");
+	        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		}
+		
+		return ResponseEntity.ok(questionLogic.deleteQuestion(id));
+	}
+	
+//QUESTIONCOMMENT CUD---------------------------------
+	
+	@RequestMapping(method = RequestMethod.POST, value = "/comments")
+	public ResponseEntity<QuestionComment> addEvalComment(@RequestBody QuestionComment comment){
+		return ResponseEntity.ok(questionLogic.createComment(comment));
+	}
+	
+	@RequestMapping(method = RequestMethod.PUT, value = "/comments/update/{id}")
+	public ResponseEntity<QuestionComment> updateEvalComments(@RequestBody QuestionComment comment, @PathVariable ("id") Integer id){
+		
+		QuestionComment currComment = questionLogic.getCommentById(id);
+		
+		if(currComment == null){
+			System.out.println("Comment with id " + id + " not found");
+	        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		}
+		
+		if(comment.getCommentText() != null){
+			currComment.setCommentText(comment.getCommentText());
+		}
+		if(comment.getQuestionEval() != null){
+			currComment.setQuestionEval(comment.getQuestionEval());
+		}
+		
+		return ResponseEntity.ok(questionLogic.updateComment(currComment));
+	}
+	
+	@RequestMapping(method = RequestMethod.DELETE, value = "/comments/delete/{id}")
+	public ResponseEntity<QuestionComment> deleteQuestionComment(@PathVariable ("id") int id){
+		
+		QuestionComment currComment = questionLogic.getCommentById(id);
+		
+		if(currComment == null){
+			System.out.println("Comment with id " + id + " not found");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)